src/ai/core/market_predictor.py

```
@@ -0,0 +1,144 @@
+                property_type=performance_data.get('type', 'unknown'),
+                trend_direction=trend_direction,
+                momentum=momentum,
+                confidence=confidence
+            )
+        except Exception as e:
+            print(f"Error analyzing property trend: {str(e)}")
+            return PropertyTrend('unknown', 0.0, 0.0, 0.0)
+
+    def _generate_trend_recommendations(
+        self,
+        property_trends: Dict[str, PropertyTrend]
```

Ruff (0.8.2)

Trailing comma missing

(COM812)